src/e2ee/RustEngine.ts

```diff
@@ -107,7 +113,7 @@ export class RustEngine {
         settings.rotationPeriodMessages = BigInt(encEv.rotationPeriodMessages);
 
         await this.lock.acquire(roomId, async () => {
-            const requests = JSON.parse(await this.machine.shareRoomKey(new RoomId(roomId), members, settings));
+            const requests = JSON.parse(await this.machine.shareRoomKey(new RoomId(roomId), members, settings)) as RequestBody[];
```

Comment:
We should be getting types from the rust-sdk on this, largely to prevent this exact error

Reply:
Does there exist a type for this yet?  The only mention of the request body properties that I could find was within a comment in the generated `index.js`/`index.d.ts`:
```
export class ToDeviceRequest {
  /** The request ID. */
  readonly id: string
  /**
   * A JSON-encoded object of form:
   *
   * ```json
   * {"event_type": …, "txn_id": …, "messages": …}
   * ```
   */
  readonly body: string
  /** Get its request type. */
  get type(): RequestType
}
```

Or is the fact that no TS type is generated for `body` what you meant by ["the TS types for the rust-sdk are likely wrong"](https://github.com/matrix-org/matrix-rust-sdk/issues/1093#issuecomment-1296957629)?